apps/homework/views.py

```diff
@@ -54,53 +54,186 @@ def get_context_data(self, **kwargs):
         return context
 
 
-class SubmissionOverView(LoginRequiredMixin, TemplateView):
+class HomeworkOverView(LoginRequiredMixin, TemplateView):
     template_name = 'homework/overview.html'
     model = Submission
 
     def get_context_data(self, **kwargs):
-        context = super(SubmissionOverView, self).get_context_data(**kwargs)
+        context = super(HomeworkOverView, self).get_context_data(**kwargs)
+        klass_pk = self.kwargs.get('klass_pk')
         try:
-            klass = Klass.objects.get(pk=self.kwargs.get('klass_pk'))
+            klass = Klass.objects.get(pk=klass_pk)
             context['klass'] = klass
         except ObjectDoesNotExist:
             raise Http404('Klass object not found')
         return context
 
+#class SubmissionMetricsView(LoginRequiredMixin, UserPassesTestMixin, TemplateView):
+#    template_name = 'homework/admin_metrics.html'
+#
+#    def test_func(self):
+#        definition_pk = self.kwargs.get('definition_pk')
+#        try:
+#            definition = Definition.objects.get(pk=definition_pk)
+#        except ObjectDoesNotExist:
+#            raise Http404('Definition object not found')
+#        if self.request.user.instructor:
+#            if self.request.user.instructor == definition.klass.instructor:
+#                print('User is instructor of class.')
+#                return True
+#        try:
+#            if self.request.user.klass_memberships.get(klass__pk=definition.klass.pk):
+#                print('User is student of class.')
+#                return True
+#        except ObjectDoesNotExist:
+#            return False
+#        return False
+
+class SubmissionDetailView(LoginRequiredMixin, UserPassesTestMixin, TemplateView):
+    template_name = 'homework/submission_detail.html'
+
+    def get_context_data(self, **kwargs):
+        context = super(SubmissionDetailView, self).get_context_data(**kwargs)
+        submission_pk = self.kwargs.get('submission_pk')
+        try:
+            submission = Submission.objects.get(pk=submission_pk)
+            context['submission'] = submission
+            context['definition'] = submission.definition
+        except ObjectDoesNotExist:
+            raise Http404('Definition object not found')
+        return context
+
+    def test_func(self):
+        submission_pk = self.kwargs.get('submission_pk')
+        try:
+            submission = Submission.objects.get(pk=submission_pk)
+        except ObjectDoesNotExist:
+            raise Http404('Definition object not found')
+        if self.request.user.is_superuser:
+            return True
+        if self.request.user.instructor:
+            if self.request.user.instructor == submission.definition.klass.instructor:
+                print('User is instructor of class.')
```

Comment:
Probably pull print, and only log when we fail, and why

Reply:
Thanks for the principle here.

Reply:
Done.